sputnikdao2/tests-ava/__tests__/upgrade.ava.ts

```diff
@@ -0,0 +1,163 @@
+// 2. add proposal for UpgradeSelf with hash of blob from #1(get it approved)
+// 3. add proposal for remove_contract_self(get it approved)
+// 4. Confirm DAO contract code_hash and returned balance
+
+workspaceWithFactory.test('basic', async (test, { alice, root, factory, dao }) => {
+		test.true(await alice.exists())
+		test.true(await root.exists())
+		test.true(await factory.exists())
+		test.true(await dao.exists())
+		test.log(await dao.view('get_config'))
+})
+
+workspaceWithFactory.test('Store DAO upgrade code in DAO via factory', async (test, { alice, root, factory, dao }) => {
+		const six_near = toYocto('6');
+		const default_code_hash = await factory.view('get_default_code_hash');
+		// const dao = await factory.makeSubAccount('upgradedao');
+		// console.log('default_code_hash', default_code_hash);
+
+		const config = {
+        name: 'upgradedao',
+        purpose: 'to test',
+        metadata: ''
+    };
+    const policy = [root.accountId];
+    const params = {
+        config,
+        policy
+    };
+
+    await root.call(
+        factory,
+        'create', {
+            name: 'upgradedao',
+            args: Buffer.from(JSON.stringify(params)).toString('base64')
+        }, {
+            attachedDeposit: toYocto('10'),
+            gas: tGas(300),
+        }
+    );
+
+		test.deepEqual(await factory.view('get_dao_list', {}), ['upgradedao.factory.test.near']);
+		
+		// 1. add proposal for store_contract_self(get it approved)
+		// --------------------------------------------------------------------
+		test.is(await dao.view('get_last_proposal_id'), 0);
+		const args = Buffer.from(`{ "code_hash": "${default_code_hash}" }`, 'binary').toString('base64');
+		
+		const proposal = {
+				proposal: {
+						description: 'Store DAO upgrade contract code blob',
+						kind: {
+								FunctionCall: {
+										receiver_id: `${factory.accountId}`,
+										actions: [
+												{
+														method_name: 'store_contract_self',
+														args: args,
+														deposit: six_near,
+														gas: tGas(220)
+												}
+										]
+								}
+						}
+				}
+		};
+
+		//Checks that the same proposal doesn't fail 
+		await root.call(dao, 'add_proposal', proposal, { attachedDeposit: toYocto('1'), gas: tGas(300) })
+		test.is(await dao.view('get_last_proposal_id'), 1);
+
+		let new_proposal: any = await dao.view('get_proposal', { id: 0 })
+
+		test.log(new_proposal);
+		test.is(new_proposal.description, 'Store DAO upgrade contract code blob');
+		test.is(new_proposal.proposer, 'test.near')
+		test.is(new_proposal.status, 'InProgress')
+		test.truthy(new_proposal.kind.FunctionCall)
+		test.is(new_proposal.kind.FunctionCall.receiver_id, `${factory.accountId}`)
+
+		await root.call(dao, 'act_proposal', { id: 0, action: 'VoteApprove' }, { gas: tGas(300) })
+		let passed_proposal_0: any = await dao.view('get_proposal', { id: 0 })
+		test.log(passed_proposal_0);
+		test.is(passed_proposal_0.status, 'Approved')
```

Comment:
Hey Trevor. The reason why this whole flow is failing is because you are not creating the DAO via the factory in the first place.

The DAO that you're using in this test is stored inside `dao.test.near`.

The factory that you're using in this test is stored inside `factory.test.near`.

Of course the test is failing, because dao should have an account which looks more like `dao.factory.test.near`. Otherwise the `store_contract_self` method inside the factory will fail with the following error:

```
        assert_eq!(factory_id, env::current_account_id().as_str(), "Wrong factory");
```

because factory_id will be `test.near` and env::current_account_id() will be `factory.test.near`